### ISSUE TITLE
Feature Request: Bookmark articles and if possible then in multiple lists

### ISSUE THREAD

**Issue by ShivamS136** [javascript-tutorial/en.javascript.info#2754]
I am learning and brushing up my JS skills by this site only, I am learning from beginning to the end. 

While reading I realized that there are some articles which I need to revisit regularly so that they are fresh in my mind, some are there which I would like to read later and some are very fabulous articles to which I would refer always when I need to understand or to explain someone the same.

So for this, I propose a feature to bookmark articles, also, if possible then in multiple lists like below:
* Favorite
* Read Later
* Important

Though only a single "Bookmark" feature will also help a lot.

We can discuss the need for such a feature here and once we are OK to have the feature we can discuss how to implement it.

**Comment by Rajeshsharma21** [comment]
> I am learning and brushing up my JS skills by this site only, I am learning from beginning to the end.
> 
> While reading I realized that there are some articles which I need to revisit regularly so that they are fresh in my mind, some are there which I would like to read later and some are very fabulous articles to which I would refer always when I need to understand or to explain someone the same.
> 
> So for this, I propose a feature to bookmark articles, also, if possible then in multiple lists like below:
> 
> * Favorite
> * Read Later
> * Important
> 
> Though only a single "Bookmark" feature will also help a lot.
> 
> We can discuss the need for such a feature here and once we are OK to have the feature we can discuss how to implement it.

yes, i also need such feature on this website. nice to see your comment on such issue.

**Comment by Rajeshsharma21** [comment]
yes, i also need such feature on this website. nice to see your comment on such issue

**Comment by ShahriarKh** [comment]
+1 for this. I'm planning to use the site as my main ref for js, and it will be a really handy option.

**Comment by iliakan** [comment]
I'm very confused about this request.

The reason is that most modern browsers, including Chrome and Firefox, allow to synchronize the profile between devices.

So one can use regular browser bookmarks. They'll be synced.

Is it that people just don't know how to sync their browser profile? Is it the only reason such request exists? Otherwise, please explain, I'm all ears.

**Comment by ShahriarKh** [comment]
Here are some reasons I can think of:
1. What if you use various browsers? For example, Chrome on device 1, Firefox on device 2 and Safari on device 3.
2. What if you don't want to login to your browser profile? Imagine you're at work or at school and don't want to sync everything with the browser you're working with. You just want to login to your `javascript.info` account and continue your progress.
3. What if you use a 3rd-party bookmark manager which is only available on some devices? e.g., it works on your computer but not on your phone.

However, the main point is not using the browser profile instead of actual site profile, but having an in-app bookmark functionality instead of struggling with browser bookmarks.

**Comment by ShahriarKh** [comment]
![image](https://user-images.githubusercontent.com/31452340/152681694-7cc628bf-7577-4722-9ebc-4efba8dae848.png)
Here's just a quick screenshot of what's in my (and possibly others') mind.
```css
.list-sub__item:nth-child(5n) .list-sub__title .list-sub__link::after {
  background: red;
  content: "📘"; /* or 🔖*/
  position: absolute;
  right: 24px;
}
```

**Comment by ShivamS136** [comment]
Additional to @ShahriarKh 
We already have sign in feature in the site, so I think adding bookmark feature in site should be easier.

Also this js the real centralisation of the data, not the browser bookmark thing, browsers can be different and bookmarks are harder to manage.

**Comment by ShivamS136** [comment]
> ![image](https://user-images.githubusercontent.com/31452340/152681694-7cc628bf-7577-4722-9ebc-4efba8dae848.png) Here's a quick screenshot of what's in my (and possibly others') mind.
> 
> ```css
> .list-sub__item:nth-child(5n) .list-sub__title .list-sub__link::after {
>   content: "📘"; /* or 🔖*/
>   position: absolute;
>   right: 24px;
> }
> ```

Actually in my mind is having a bookmark button on the post, on this page and a separate page with all bookmarked articles or a filter on this page for bookmarked articles.

**Comment by ShahriarKh** [comment]
Of course. There should be a button somewhere in the article itself so you can toggle the bookmark.
In addition to bookmarks, we can use these emojis (icons) for other requested features:
![image](https://user-images.githubusercontent.com/31452340/152682124-64015b6d-895e-48b4-b197-dd05b62f7f10.png)
📘 or 🔖: Bookmarked
✅: Studied (see #1913, which I'm really interested in)
🔄: Updated (Since your last visit/Since the last time you hit "studied"/...)

**Comment by iliakan** [comment]
@bezart32, what you think?

**Comment by iliakan** [comment]
We need a vote on features =) 

Any service for that, with ability to suggest new features?

**Comment by ShahriarKh** [comment]
There are bunch of those. I took a quick look here https://alternativeto.net/software/productroad/ and you can look too; but "convas" offers a free plan for open-source projects: https://convas.io/open-source.
Another alternative is to use GitHub discussions.

**Comment by ShahriarKh** [comment]
Wanna introduce it in Discord chat and also put a link in the website?

**Comment by iliakan** [comment]
Let's put several issues there first.

**Comment by iliakan** [comment]
Is it a convenient place to describe them and vote?

**Comment by ShahriarKh** [comment]
I think GitHub discussions is better, but we need a "general" repo rather than "en.javascript.info" so different people can suggest things. Something like "javascript.info website" or "javascript.info". What do you think?

**Comment by iliakan** [comment]
We have this: https://github.com/javascript-tutorial/server

Why are discussions better (despite keeping it on github)?

**Comment by ShivamS136** [comment]
> We have this: https://github.com/javascript-tutorial/server
> 
> Why are discussions better (besides keeping it on github)?

I don't think we should have another platform to discuss the features to implement on javascript.info, being developed at github.com. For upvotes, we can use Github discussions or 👍 emoji counts on an issue. We should keep it simple. If discussions need to be private then it's a different scenario.

**Comment by iliakan** [comment]
I guess, makes sense to make discussions here. And if other languages want something specific, then they can add a discussion there.

**Comment by iliakan** [comment]
I see a fundamental flaw with such voting.

Let's say feature A was added a week ago, and feature B was added a year ago.

Surely, feature B will have more votes, just because the voting took longer.

To judge fairly, we must compare something like "votes last month".

Is it possible to get such metric in github?

**Comment by ShahriarKh** [comment]
Yes it does:
![image](https://user-images.githubusercontent.com/31452340/153924759-b595e855-37a0-4cc0-9186-49d32572ba38.png)

**Comment by iliakan** [comment]
No, it shows all votes, no matter which top you choose. Try it.

**Comment by ShahriarKh** [comment]
This is because we created all of them on the same day :trollface: 
See it in a repo with a large amount of posts:
https://github.com/tannerlinsley/react-query/discussions?discussions_q=sort%3Atop+created%3A%3E%3D2022-02-14

**Comment by aayush-goyal** [comment]
Any updates on this? This is actually a quite useful feature IMO. I can pick this up on implementation side.
